types/ramda/index.d.ts

```diff
@@ -1447,8 +1449,8 @@ declare namespace R {
          * Returns a function that when supplied an object returns the indicated property of that object, if it exists.
          * Note: TS1.9 # replace any by dictionary
          */
-        prop<T>(p: string, obj: any): T;
-        prop<T>(p: string): (obj: any) => T;
+        prop<P extends string, T>(p: P, obj: Record<P, T>): T;
+        prop<P extends string>(p: P): <T>(obj: Record<P, T>) => T;
```

Comment:
What do you think of the following type instead?

```ts
prop<P extends string>(p: P): <T extends Record<P, any>(obj: T): Pick<T, P>;
```

Reply:
AFAIK the they’re equivalent; is there a reason to prefer that? I find yours a little more complex to read.

Reply:
Wait, you mean `O[P]` instead of `Pick<T, P>`, don’t you?

Reply:
Oh, yes, probably. Sorry, I don't use it that much.

Reply:
They're actually not exactly equivalent. With `Record<P, T>`, you're expecting an object that is homogeneous for some well-known `T` (meaning that if you have `{x: number, y: string}`, you will always get `number | string`. If you make `T` the object type, and then use an indexed access, you've delayed the evaluation of the return type until `T` is fully known, giving you a more accurate `prop`.

Reply:
@DanielRosenwasser ok, but in this case `P` can't be a union of multiple literals, since it's determined by a single `p: P`. So `Record<P, T>` amounts to just `{ [p]: T }`.

Reply:
Well, I guess you could always explicitly widen `P` to a union of literals or even `string`. Fair point, I think you're right.